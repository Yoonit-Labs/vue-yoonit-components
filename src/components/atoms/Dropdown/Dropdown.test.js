/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

import { shallowMount } from '@vue/test-utils'
import YooDropdown from '@/components/atoms/Dropdown/Dropdown.vue'
import PropsConfig from '@/components/atoms/Dropdown/Dropdown.config'

const classBlock = 'yoo-dropdown'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooDropdown, {
    slots: { default: SlotText },
    propsData: { options: [{ label: 'YooDropDown 1', value: '1' }, { label: 'YooDropDown 2', value: '2' }] }
  })
}

describe('YooDropdown Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component Dropdown HTML', () => {
    expect(wrapper.find('.yoo-dropdown').exists()).toBe(true)
    expect(wrapper.find('.yoo-dropdown__select').exists()).toBe(true)
    expect(wrapper.find('.select').exists()).toBe(true)
    expect(wrapper.find('.select__text').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooDropdown.props.size.default)).toBe(true)
      })
      PropsConfig.size.options.forEach(size => {
        it(`Includes size class: .${classBlock}--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}--${size}`).exists()).toBe(true)
        })
      })
    })
  })

  describe('Events', () => {
    describe('Click', () => {
      it('Options are loaded by clicking on select', async () => {
        await wrapper.find('.yoo-dropdown__select').trigger('click')
        expect(wrapper.find('.yoo-dropdown__options').exists()).toBe(true)
      })
      it('Emits Click Event', async () => {
        await wrapper.find('.yoo-dropdown__select').trigger('click')
        expect(wrapper.find('.yoo-dropdown__options').exists()).toBe(true)
        await wrapper.find('.yoo-dropdown__options .option').trigger('click')
        expect(wrapper.emitted()).toHaveProperty('onClick')
      })
    })
  }) // describe Events
})
