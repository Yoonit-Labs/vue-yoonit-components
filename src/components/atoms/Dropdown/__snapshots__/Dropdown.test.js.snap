// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooDropdown Component Matches Snapshot 1`] = `
<div class="yoo-dropdown yoo-dropdown yoo-dropdown--medium">
  <div class="yoo-dropdown__select">
    <div class="select">
      <p class="select__text">YooDropDown 1</p>
    </div>
  </div>
  <!---->
</div>
`;
